deeplearning4j-nearestneighbors-parent/deeplearning4j-nearestneighbor-server/src/main/java/org/deeplearning4j/nearestneighbor/server/NearestNeighbor.java

```diff
@@ -28,8 +28,15 @@
             List<DataPoint> add = new ArrayList<>();
             List<Double> distances = new ArrayList<>();
             tree.search(input, record.getK(), add, distances);
-            for (DataPoint dataPoint : add) {
-                results.add(new NearestNeighborsResult(dataPoint.getIndex()));
+
+            if (add.size() != distances.size()) {
+                throw new IllegalStateException(
+                        String.format("add.size(%d) != distances.size(%d)",
```

Comment:
While I applaud the use of format, this syntax may lead readers to think the value of `add.size()` is the argument to size. 

How about "add.size == %d != %d == distances.size" ?